### PR TITLE
Cf 920 finalize scripts

### PR DESCRIPTION
**Pull request by john-chrosniak**
# PR Details
## Description

This PR includes fixes for plots and metrics that will be used for C1T verification testing. Specifically, the fixes include

1. Processing route graphs using the specified nodes/edges instead of fitting a spline through points
2. Adding a buffer to allow the vehicle to accelerate before checking if it is tracking its target speed
3. Adding a check that both incoming MOMs and Rviz goals are reached appropriately

## Related GitHub Issue

NA

## Related Jira Key

[CF-920](https://usdot-carma.atlassian.net/browse/CF-920)

## Motivation and Context

Adjustments to the data analysis scripts for C1T are needed to account for route graphs that contain branches (i.e. split between standard exit and inspection lane). Additional updates will likely be needed as data analysis with the data collected during integration testing begins.

## How Has This Been Tested?

Tested using data collected during integration testing.

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

[CF-920]: https://usdot-carma.atlassian.net/browse/CF-920?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by john-chrosniak** [comment]
crosstrack_vs_downtrack
route_driven

File: src/carma-1-tenth/check_distance_to_arrival.py [hunk 1]
```diff
@@ -22,23 +22,23 @@ def check_distance_to_arrival(bag_dir):
...
     # Open bag
     reader, type_map = open_bagfile(bag_dir, topics=[goal_topic, ack_topic], storage_id=storage_id)
     # Gather number of messages on each topic
     topic_count_dict = {entry["topic_metadata"]["name"] : entry["message_count"] for entry in metadata_dict["topics_with_message_count"]}
     # Number of goal messages must equal number of acks
-    if topic_count_dict[goal_topic] != topic_count_dict[ack_topic]:
-        print("Number of goal messages (%d) does not equal number of ack messages (%d)".format(topic_count_dict[goal_topic], topic_count_dict[ack_topic]))
-        return np.array([np.inf])
+    if topic_count_dict[goal_topic] + topic_count_dict[rviz_topic] != topic_count_dict[ack_topic]:
+        print("Number of goal messages ({0}) does not equal one less than number of ack messages ({1})".format(topic_count_dict[goal_topic] + topic_count_dict[rviz_topic], topic_count_dict[ack_topic]))
```
Comment (adev4a) [inline src/carma-1-tenth/check_distance_to_arrival.py:34]: Based on this message it seems like the conditional above should be checking for equality with `topic_count_dict[ack_topic] - 1`

File: src/carma-1-tenth/check_distance_to_arrival.py [hunk 2]
```diff
@@ -49,6 +49,9 @@ def check_distance_to_arrival(bag_dir):
                 # Store goal position
                 goal_positions[goal_position_count] = [strategy_params["destination"]["longitude"], strategy_params["destination"]["latitude"]]
                 goal_position_count += 1
+            elif topic == rviz_topic:
+                goal_position_count[goal_position_count] = [msg.pose.position.x, msg.pose.position.y]
```
Comment (adev4a) [inline src/carma-1-tenth/check_distance_to_arrival.py:53]: Is this supposed to be `goal_positions[goal_position_count]`?

File: src/carma-1-tenth/run_c1t_metric_analysis.py [hunk 3]
```diff
@@ -52,7 +52,20 @@ def test_C1T_06_maintains_speed_on_straights(self):
         # The C1T CMV achieves and maintains its target speed with a tolerance of 0.2 m/s (excluding turns)
         velocities, target_velocities = plot_vehicle_speed(self.bag_dir, show_plots=False)
         max_target_speed = np.max(target_velocities)
-        speeds_on_straights = velocities[target_velocities == max_target_speed]
+        wait_for_vehicle_to_accelerate = 0
+        relevant_velocity_idxs = []
+        # Add a buffer of ~3 seconds to allow the vehicle to accelerate before it is expected to reach its maximum speed
```
Comment (adev4a) [inline src/carma-1-tenth/run_c1t_metric_analysis.py:57]: Is wait_for_vehicle_to_accelerate the buffer being referenced here? Comment here says add buffer of 3 seconds, but only 1 second is added below

File: src/carma-1-tenth/check_distance_to_arrival.py (hunk 1 above)
Comment (john-chrosniak) [reply src/carma-1-tenth/check_distance_to_arrival.py]: Yes the error message was outdated, just fixed

File: src/carma-1-tenth/check_distance_to_arrival.py (hunk 2 above)
Comment (john-chrosniak) [reply src/carma-1-tenth/check_distance_to_arrival.py]: Yup, good catch!

File: src/carma-1-tenth/run_c1t_metric_analysis.py (hunk 3 above)
Comment (john-chrosniak) [reply src/carma-1-tenth/run_c1t_metric_analysis.py]: Yes that is the buffer, I forgot to push the changes that increased the buffer. The buffer is now ~2 seconds and I updated the comment.

**Review by willjohnsonk — DISMISSED** [review]

**Review by willjohnsonk — APPROVED** [review]

**Pull request merged**